ds/beis_indicators/trademarks/make_trademarks.py

```diff
@@ -0,0 +1,253 @@
+    """
+    Function to collect the trademark data
+    Args:
+        url (str) is the link for the data
+    """
+    if  os.path.exists(f"{project_dir}/data/raw/trademarks/OpenDataDomestic.txt"):
+        logger.info("Already collected the data")
+
+    else:
+        req = requests.get(url)
+
+        # Extracts the text file (in raw)
+        logger.info("Saving the raw data")
+
+        ZipFile(BytesIO(req.content)).extract(
+            "OpenDataDomestic.txt", path=f"{project_dir}/data/raw/trademarks/"
+        )
+
+    if os.path.exists(f"{project_dir}/data/raw/trademarks/trademarks_processed.df"):
+
+        logger.info("Already processed the data")
+        text_df = pd.read_csv(f"{project_dir}/data/raw/trademarks/trademarks_processed.df")
+
+    else:
+        logger.info("Creating dataframe")
+
+        # Reads the text file
+        with open(
+            f"{project_dir}/data/raw/trademarks/OpenDataDomestic.txt",
+            "r",
+            encoding="utf-16",
+        ) as infile:
+
+            # There are a few corrupted lines with errors (more than 62 fields)
+            text = infile.read().split("\n")
+
+            text_clean = [x.split("|") for x in text if len(x.split("|")) == 62]
+
+            bad_lines = len(text) - len(text_clean)
+            bad_lines_rate = np.round(bad_lines / len(text), 3)
+
+            logger.info(
+                f"We found {bad_lines} bad lines, {100*bad_lines_rate}% of the total"
+            )
+
+        # Creates the dataframe (note that the first row are the headers)
+        text_df = pd.DataFrame(text_clean[1:], columns=text_clean[0])
+
+        # Tidy the columns
+        text_df.columns = _tidy_columns(text_df.columns)
+
+        text_df.to_csv(f"{project_dir}/data/raw/trademarks/trademarks_processed.df",
+                       index=False)
+
+    return text_df
+
+def geocode_trademarks(df, geo_code=['long','lat']): ##CHANGETHIS
+    """
+    This function reverse geocodes a trademark df using the postcode of the applicant.
+    It returns the lat and lon for each trademark
+
+    Args:
+        df (df) is a dataframe with organisation postcodes
+        geo_code (str) is the geocode in the NSPL database that we want to use
+    """
+    # We have trailing spaces in the postcodes
+    df["postcode"] = [
+        x.strip() if pd.isnull(x) is False else np.nan for x in df.postcode
+    ]
+
+    # Read nspl
+    nspl = pd.read_csv(f"{project_dir}/data/raw/nspl/Data/NSPL_FEB_2020_UK.csv") 
+
+    # The trademark dataset only provides information for the first part of the postcode
+    # We split the nspl postcodes to merge on a smaller dataset
+    nspl["pcds_1st"] = nspl["pcds"].apply(lambda x: x.split(" ")[0])
+
+    nspl_short = nspl.drop_duplicates("pcds_1st")[["pcds_1st"]+geo_code]
+
+    merged = pd.merge(df, nspl_short, left_on="postcode", right_on="pcds_1st")
+
+    return merged
+
+def make_reverse_geocode_lookup(df,nuts_years=[2010,2013,2016],coords=['long','lat']):
```

Comment:
There is already a `NUTS_YEARS` variable there. 

We are only using these because we are focusing on trademarks published after 2010 so older versions of NUTS aren't relevant.